### DISCUSSION TITLE
How to build a multi-scale model as shown in the diagram?

### DISCUSSION THREAD

**Discussion by aozakiccaoko** [neperfepx/neper#823] [Q&A] [answered]
Dear Neper team, I would like to create a model similar to the one shown in the figure, with the yellow part indicating the grains that need to be inserted. I would greatly appreciate your guidance on this matter.
![4b8a970d522b93420fb8841b572d4c6](https://github.com/neperfepx/neper/assets/105437431/6a8ad899-42c0-4d23-a4ea-8e89187f077e)

**Comment by rquey** [comment, marked as answer]
This cannot be done really easily.  I would generate a polycrystal for the green grain structure and set some of the elements to "grain 2" (assuming your using a mesh).  If you're using a tesr, you can manually "convert" some of the voxels.

Another approach would be to use a multiscale tessellation, but it won't be very easy with ~cubic inserts, and it will add many constraints...
